tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/ExemplarMetadataProvider.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics.registry;
+
+import javax.annotation.Nullable;
+
+public interface ExemplarMetadataProvider<T> {
+    @Nullable
+    T collect(long value);
```

Comment:
Since we've added the `LongExemplar<T>` type, I think we can remove the `long value` arg here -- the collected metadata and value will be combined later on (unless you have a case in mind where the value is necessary for metadata collection -- otherwise it becomes easy to accidentally hold the same value on heap in two places unnecessarily).

Reply:
You're right and I don't have a case in mind for using the value in metadata collection, just a mind blip.